pkg/processor/template_utils.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/conduitio/conduit/pkg/record"
+)
+
+// TemplateUtils provides methods to parse, execute go templates for provided records, and return the boolean value of the output.
+type TemplateUtils struct {
```

Comment:
Last but not least, some naming nitpicks. `TemplateUtils` doesn't really tell me what this type is used for. IIUC we're creating a type that knows how to evaluate a condition attached to a processor and returns `true` or `false`. Well, then how about `processorCondition`? Since we're in package `processor` we could even simply use `condition` (although that's debatable, as the type won't be used outside of this package, so it will never be referred to as `processor.condition`).

And then, what do you do with a condition? You evaluate it. So I would rename `execute` to `Evaluate`. Notice I made it public, as it's meant to be used by code outside of this type.

Reply:
thanks for all the suggestions!
I went with `processorCondition`, thought it made more sense, especially since it won't be used outside the package (as you mentioned).